pkg/destroy/libvirt/libvirt.go

```diff
@@ -83,145 +96,124 @@ func (o *ClusterUninstaller) Run() (*types.ClusterQuota, error) {
 // additional nodes after the initial list call.  We continue deleting
 // domains until we either hit an error or we have a list call with no
 // matching domains.
-func deleteDomains(conn *libvirt.Connect, filter filterFunc, logger logrus.FieldLogger) error {
+func deleteDomains(virt *libvirt.Libvirt, filter filterFunc, logger logrus.FieldLogger) error {
 	logger.Debug("Deleting libvirt domains")
 	var err error
 	nothingToDelete := false
 	for !nothingToDelete {
-		nothingToDelete, err = deleteDomainsSinglePass(conn, filter, logger)
+		nothingToDelete, err = deleteDomainsSinglePass(virt, filter, logger)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func deleteDomainsSinglePass(conn *libvirt.Connect, filter filterFunc, logger logrus.FieldLogger) (nothingToDelete bool, err error) {
-	domains, err := conn.ListAllDomains(0)
+func deleteDomainsSinglePass(virt *libvirt.Libvirt, filter filterFunc, logger logrus.FieldLogger) (nothingToDelete bool, err error) {
+	domains, _, err := virt.ConnectListAllDomains(1, 0)
```

Comment:
nit: Using a more speaking flag like `ConnectListDomainsActive` could be slightly more readable

Reply:
The 1 here is the `NeedResults` parameter (sometimes you just need a count; this isn't one of those times). No flags are being passed, hence `0` (there's no other constant defined for empty flags). The libvirt-go `conn.ListAllDomains()` function took only a flags argument.